src/soroban/assembled_transaction.ts

```diff
@@ -0,0 +1,717 @@
+import type { ContractClientOptions, XDR_BASE64 } from ".";
+import {
+  Account,
+  BASE_FEE,
+  Contract,
+  Operation,
+  SorobanRpc,
+  StrKey,
+  TimeoutInfinite,
+  TransactionBuilder,
+  authorizeEntry,
+  hash,
+  xdr,
+} from "..";
+import type { Memo, MemoType, Transaction } from "..";
+
+type Tx = Transaction<Memo<MemoType>, Operation[]>;
+
+type SendTx = SorobanRpc.Api.SendTransactionResponse;
+type GetTx = SorobanRpc.Api.GetTransactionResponse;
+
+/**
+ * Error interface containing the error message
+ */
+interface ErrorMessage {
+  message: string;
+}
+
+interface Result<T, E extends ErrorMessage> {
+  unwrap(): T;
+  unwrapErr(): E;
+  isOk(): boolean;
+  isErr(): boolean;
+}
+class Ok<T> implements Result<T, never> {
+  constructor(readonly value: T) {}
+  unwrapErr(): never { throw new Error("No error") }
+  unwrap() { return this.value }
+  isOk() { return true }
+  isErr() { return false }
+}
+class Err<E extends ErrorMessage> implements Result<never, E> {
+  constructor(readonly error: E) {}
+  unwrapErr() { return this.error }
+  unwrap(): never { throw new Error(this.error.message) }
+  isOk() { return false }
+  isErr() { return true }
+}
+
+export type MethodOptions = {
+  /**
+   * The fee to pay for the transaction. Default: BASE_FEE
+   */
+  fee?: number;
+};
+
+export type AssembledTransactionOptions<T = string> = MethodOptions &
+  ContractClientOptions & {
+    method: string;
+    args?: any[];
+    parseResultXdr: (xdr: xdr.ScVal) => T;
+  };
+
+export const NULL_ACCOUNT =
+  "GAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAWHF";
+
+export class AssembledTransaction<T> {
+  public raw?: Tx;
+  private simulation?: SorobanRpc.Api.SimulateTransactionResponse;
+  private simulationResult?: SorobanRpc.Api.SimulateHostFunctionResult;
+  private simulationTransactionData?: xdr.SorobanTransactionData;
```

Comment:
Aren't the latter two already part of the first one?

Reply:
We would prefer to only have the first one, but couldn't figure out serialization/deserialization if we only had `simulation`. Adding the other two was a workaround. Maybe you could help us figure out `simulation` serialization!